src/pages/index.tsx

```diff
@@ -1,13 +1,19 @@
 import TokenStatsDisplay from "@components/TokenStatsDisplay";
 import { HomeTitle } from "@components/HomeTitle";
 import { SearchBar } from "layouts/components/searchbar/SearchBar";
+import { GetServerSidePropsResult, InferGetServerSidePropsType } from "next";
 import GroupStatisticCard from "@components/GroupStatisticCard";
-import LatestDataTable from "@components/LatestDataTable";
+import LatestDataTable, { RowData } from "@components/LatestDataTable";
 import LatestDataApi from "@api/LatestDataApi";
+import { MAIN_LATEST_TRANSACTION_URL } from "@api/index";
+import dayjs from "dayjs";
+import { TransactionType } from "../mockdata/TransactionData";
 
-export default function Home() {
+export default function Home({
+  latestTransactions,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
   const latestBlocks = LatestDataApi.useLatestBlocks();
-  const latestTransactions = LatestDataApi.useLatestTransactions();
+  // const data = LatestDataApi.useLatestTransactions();
```

Comment:
I just realised we can actually reuse the `LatestDataApi.ts` file, we just have to modify it a bit and pass the transaction/block list on this line

Pros:  index.tsx page will be a bit cleaner as most of the data transformation can be done on the api file
Cons: extra layer when debugging

Thoughts? @JoshuaC817 @fullstackninja864 @pierregee

Reply:
hmm since we are using SSR, not sure if the same implementation could work? i'll try it out and let you guys know